layer_metadata_restorer.go

```diff
@@ -14,19 +14,24 @@ import (
 //go:generate mockgen -package testmock -destination testmock/layer_metadata_restorer.go github.com/buildpacks/lifecycle LayerMetadataRestorer
 type LayerMetadataRestorer interface {
 	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata) error
+	CacheIsValid(buildpackID, cachedLayerSha string, layer BpLayer) (bool, error)
```

Comment:
Yeah, I thought about it but since it's still in progress and I wanted to make sure that we're all ok with this change, I didn't add it yet. Some of my thoughts appear [here](https://github.com/buildpacks/lifecycle/pull/624#discussion_r643423966).

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/624/commits/f584e397d9e441ee1860957da6c18e023d65a89d.